src/symbolic/symbolic_choice.ml

```diff
@@ -132,25 +135,19 @@ end = struct
       *)
     type ('a, 'wls) work_queue = ('a, 'wls) Schedulable.t Wq.t
 
-    type 'a res_queue = 'a Wq.t
-
-    type ('a, 'wls) t =
-      { work_queue : ('a, 'wls) work_queue
-      ; res_writer : 'a res_queue
-      }
+    type ('a, 'wls) t = { work_queue : ('a, 'wls) work_queue }
 
     let init_scheduler () =
       let work_queue = Wq.init () in
-      let res_writer = Wq.init () in
-      { work_queue; res_writer }
+      { work_queue }
 
     let add_init_task sched task = Wq.push task sched.work_queue
 
-    let rec work wls sched =
+    let rec work wls sched callback =
       let rec handle_status (t : _ Schedulable.status) sched =
         match t with
         | Stop -> ()
-        | Now x -> Wq.push x sched.res_writer
+        | Now x -> callback x
```

Comment:
Is there actually any value in having all the callback mechanism instead of simply matching on `x` here to push into the queue only when it is not `EVal (Ok ())` ?

Reply:
Yes, because we can't access the type here